### PR TITLE
fix(design): allow media gallery to be registered when they are initialized

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/design/src/molecules/media-gallery/media-gallery.component.ts [hunk 1]
```diff
@@ -31,7 +33,11 @@ export class DaffMediaGalleryComponent implements DaffMediaGalleryRegistration {
...
 	@Input() name = `${uniqueGalleryId}`;
 
-	constructor() {
+	constructor(private registry: DaffMediaGalleryRegistry) {
 	  uniqueGalleryId++;
 	}
+
+	ngOnInit() {
+	  this.registry.add(this);
+	}
```
Comment (damienwebdev) [inline libs/design/src/molecules/media-gallery/media-gallery.component.ts:42]: I think we should similarly remove the registry onDestroy too.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

**Comment by xelaint** [comment]
@lderrickable can you please help me write tests for the updates I just made in the `temp` commit?

**Comment by lderrickable** [comment]
> @lderrickable can you please help me write tests for the updates I just made in the `temp` commit?

added

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**